src/Domain/CoordinateMaps/CubicScale.cpp

```diff
@@ -18,37 +23,83 @@
 #include "Utilities/ConstantExpressions.hpp"
 #include "Utilities/DereferenceWrapper.hpp"
 #include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/Gsl.hpp"
 #include "Utilities/MakeWithValue.hpp"
+#include "Utilities/StdArrayHelpers.hpp"
+#include "Utilities/TypeTraits.hpp"
 
 namespace domain {
 namespace CoordMapsTimeDependent {
 
-CubicScale::CubicScale(const double outer_boundary) noexcept
-    : outer_boundary_(outer_boundary) {
+template <size_t Dim>
+CubicScale<Dim>::CubicScale(const double outer_boundary,
+                            std::string function_of_time_name_a,
+                            std::string function_of_time_name_b) noexcept
+    : f_of_t_a_(std::move(function_of_time_name_a)),
+      f_of_t_b_(std::move(function_of_time_name_b)),
+      outer_boundary_(outer_boundary),
+      functions_of_time_equal_(f_of_t_a_ == f_of_t_b_) {
   if (outer_boundary_ <= 0.0) {
     ERROR("For invertability, we require outer_boundary to be positive.\n");
   }
 }
 
+template <size_t Dim>
 template <typename T>
-std::array<tt::remove_cvref_wrap_t<T>, 1> CubicScale::operator()(
-    const std::array<T, 1>& source_coords, const double time,
+std::array<tt::remove_cvref_wrap_t<T>, Dim> CubicScale<Dim>::operator()(
+    const std::array<T, Dim>& source_coords, const double time,
     const std::unordered_map<
         std::string, std::unique_ptr<domain::FunctionsOfTime::FunctionOfTime>>&
         map_list) const noexcept {
   const auto a_of_t = map_list.at(f_of_t_a_)->func(time)[0][0];
```

Comment:
I had to figure out what the square brackets are doing. `func(time)` returns a std::array with just one element, I guess for consistency with func_and_deriv() and func_and_2_derivs()...so that's the first [0]. And then, func(time) returns a DataVector of the components of the function. Since here a and b are scalars, that's the second [0].

Not sure if a comment would make sense here, but somewhere it would be nice to have an intro to using FunctionsOfTime.

Reply:
I can change the `auto` to `double,` but I don't think we should explain how functions of time work everywhere we use them

Reply:
I agree...mostly I was thinking this would fit in a tutorial somewhere, if there isn't already one. I mostly wrote this comment to solidify my understanding. No specific change suggested here